platforms/core-runtime/base-services/src/main/java/org/gradle/internal/exceptions/CompilationFailedIndicator.java

```diff
@@ -16,5 +16,10 @@
 
 package org.gradle.internal.exceptions;
 
+import javax.annotation.Nullable;
+
 public interface CompilationFailedIndicator extends NonGradleCause {
+
+    @Nullable
+    String getDiagnosticCounts();
```

Comment:
💅 The name of this makes me think it is going to return structured data containing the counts.  Maybe `reportDiagnosticSummary()` or something might be a better name.

Reply:
getDiagnosticSummary or getReportDiagnosticSummary?